resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -381,4 +381,42 @@ object KubernetesUtils extends Logging {
       }
     }
   }
+
+  /**
+   * This function builds the EnvVar objects for each key-value env.
+   */
+  @Since("3.4.0")
+  def buildEnvVars(env: Map[String, String]): Seq[EnvVar] = {
+    if (env.isEmpty) {
+      Seq.empty
+    } else {
+      env.filter( env => env._2 != null)
+         .map { env =>
+           new EnvVarBuilder()
+             .withName(env._1)
+             .withValue(env._2)
+             .build()
+      }
+    }.toSeq
+  }
+
+  /**
+   * This function builds the EnvVar objects for each field ref env.
+   */
+  @Since("3.4.0")
+  def buildEnvVarsWithFieldRef(env: Seq[(String, String, String)]): Seq[EnvVar] = {
+    if (env.isEmpty) {
```

Comment:
ditto. This looks like redundant and not `Scalatic`.

Reply:
`if` was removed in https://github.com/apache/spark/pull/35886/commits/cfb25897c87d7fef0ead07067b7194d60989851f